### PR TITLE
Adding pending deployment events and in progress monitor unit tests

### PR DESCRIPTION
**Pull request by narehayrapetyan**
#### Summary

Creates events for pending deployments. The events are not being acted on yet (coming in a diferent PR)

#### Testing

- [x] daemon-scheduler binary built locally and unit-tests pass (`cd daemon-scheduler; make; cd ../`)
- [x] daemon-scheduler build in Docker succeeds (`cd daemon-scheduler; make release; cd ../`)
- [x] daemon-scheduler end to end tests pass

New tests cover the changes:  yes

#### Licensing

This contribution is under the terms of the Apache 2.0 License: Yes

### CODE REVIEW

File: daemon-scheduler/pkg/engine/dispatcher_test.go [hunk 1]
```diff
@@ -30,8 +30,7 @@ import (
 )
 
 const (
-	clusterArn      = "cluster-arn"
-	environmentName = "env-name"
+	clusterArn = "cluster-arn"
```
Comment (poojamaiya) [inline daemon-scheduler/pkg/engine/dispatcher_test.go]: Can we change this to clusterARN if it isnt too much work?

File: daemon-scheduler/pkg/engine/monitor.go [hunk 2]
```diff
@@ -78,3 +110,22 @@ func (m monitor) runOnce() error {
 
 	return nil
 }
+
+func (m monitor) runPendingOnce() error {
+	environments, err := m.environment.ListEnvironments(m.ctx)
+	if err != nil {
+		return errors.New("Could not retrieve environments")
```
Comment (poojamaiya) [inline daemon-scheduler/pkg/engine/monitor.go]: Adding context here might be useful.. `Could not retrieve environments while running the pending monitor`.

File: daemon-scheduler/pkg/engine/monitor_test.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+	mocks "github.com/blox/blox/daemon-scheduler/pkg/mocks"
+	"github.com/blox/blox/daemon-scheduler/pkg/types"
+	"github.com/golang/mock/gomock"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+type MonitorTestSuite struct {
```
Comment (poojamaiya) [inline daemon-scheduler/pkg/engine/monitor_test.go]: Can we also add tests to check the cts.Done() cases for both the monitors?

File: daemon-scheduler/pkg/engine/monitor.go [hunk 4]
```diff
@@ -48,13 +50,18 @@ func NewMonitor(
 	}
 }
 
-func (m monitor) InProgressMonitorLoop() {
-	ticker := time.NewTicker(inProgressMonitorTickerDuration)
+func (m monitor) InProgressMonitorLoop(tickerDuration time.Duration) {
```
Comment (poojamaiya) [inline daemon-scheduler/pkg/engine/monitor.go]: Why this change when you're passing in the same constant in run.go?

Comment (narehayrapetyan) [inline daemon-scheduler/pkg/engine/monitor.go]: because if the ticker duration is hard coded it cannot be modified in unit tests and they end up taking 10 secs per test, which is unnecessary and makes build very long.

File: daemon-scheduler/pkg/engine/monitor.go (hunk 2 above)
Comment (narehayrapetyan) [inline daemon-scheduler/pkg/engine/monitor.go]: sure

File: daemon-scheduler/pkg/engine/monitor_test.go (hunk 3 above)
Comment (narehayrapetyan) [inline daemon-scheduler/pkg/engine/monitor_test.go]: The tests actually do hit ctx.Done() because that's how you break out of the for loop. If they didn't, the for loop would never break and the tests would not finish
